config/initializers/5_telemetry.rb

```diff
@@ -125,4 +125,5 @@
     'Widget',
   ],
 }
+allowed_labels.merge!(SETTINGS[:telemetry][:allowed_labels]) if SETTINGS[:telemetry] and SETTINGS[:telemetry][:allowed_labels]
```

Comment:
Note this is a merge, not a deep merge. So it replaces all keys. Did you intend:
```suggestion
allowed_labels.deep_merge!(SETTINGS[:telemetry][:allowed_labels]) if SETTINGS.dig(:telemetry, :allowed_labels)
```

Reply:
Good points, I am still ruby newbie :)